### DISCUSSION TITLE
combine camu.imubu.control and camu.target and simplify camu.select?

### DISCUSSION THREAD

**Discussion by diemoschwarz** [Ircam-RnD/catart-mubu#81] [Bug, Issues, and Suggestions]
- camu.imubu.control is _always_ followed by camu.target (and needs the same 2nd-to-last inlet connection)
- they could be combined (camu.target inside camu.imubu.control), since camu.imubu.control is used for specifically getting imubu to work
- (camu.target can even be made to work when inadvertently left after camu.imubu.control)

![image](https://user-images.githubusercontent.com/378151/223679481-57f833a8-38f7-4eb0-bce8-fe119b62a217.png)

**Comment by diemoschwarz** [comment]
- looking further, camu.target was also used in camu.select (and camu.voice, etc.)
- if we incorporate camu.target into camu.imubu.control, we can simplify camu.select to have only 2 inlets!!!
  - this would remove the cringeworthy decision which of the first 2 "control" inlet to use
- this means repatching in a few tutorials (3, 4, 5, 13) and examples, tell me if you're working on any of these right now, @einbond @christophertrapani

**Comment by christophertrapani** [comment]
I'm not working on those at the moment, Diemo.

I'm wondering how far into this we want to go, because strictly speaking:
there's also an inconsistency with analysis modules. Some tutorial patches
use "p corpus-analysis" (see #7) and some use "p descriptor-analysis" (see
#12); should these be standardized too (apart from the segmentation
tutorials which introduce them)?

**Comment by einbond** [comment]
Diemo, not working on those tutorials so please feel free to make the change!

Chris, if we can fix inconsistencies like that then let’s.

**Comment by diemoschwarz** [comment]
> some use "p corpus-analysis" and some use "p descriptor-analysis"

Shouldn't all of them use camu.analysis?

**Comment by diemoschwarz** [comment]
done with BIG commit f6cb8ea7848c477b51105e0f64952bd98ede5417

**Comment by diemoschwarz** [comment]
dang, we forgot to think about backwards compatibility.  `camu.select` is around since oct 2019.  Anyone using it will have to repatch...
Alternatively, we could keep the 2 middle inlets and just not use them in the examples (or say it is an "alternative use").

**Comment by diemoschwarz** [comment]
I fell into the trap myself, having used camu.select in other patches that suddenly stopped working...
I propose to undo the changes but keep the simplified tuts.
